### PR TITLE
Update glide.yaml to use current client-go

### PR DESCRIPTION
**Pull request by stu-gott**
Update all modules to reflect new k8s.io structure.

*NOTE*: This PR requires go 1.8 or newer due to changes in Kubernetes.

Signed-off-by: Stu Gott <sgott@redhat.com>

### CODE REVIEW

**Comment by berrange** [comment]
This will prevent people continuing to use Fedora 25 - they'll have to update to Fedora 26.

I've no particular objection to that, since as I wrote on the mailing list, we should aim to target latest versions only, but I wonder if it perhaps a little too soon to expect everyone to have finished the update to F26 - its only been 1 week since release

**Comment by rmohr** [comment]
In my opinion, installing go in a non-packaged form is acceptable. I quite regularly switch between go versions (like python switches with tox and virtualenv). [gimme](https://github.com/travis-ci/gimme) is very helpful there, and very stable since it is used by travis to set up go in travis-ci.

Installing a recent version of go is as simple as running
```
eval "$(curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | GIMME_GO_VERSION=stable bash)"
```

**Comment by fabiand** [comment]
Would be valuable if the readme then reflects these new requirements, including the suggestion by Roman of how to use a specific version.

File: docs/getting-started.md [hunk 1]
```diff
@@ -26,6 +26,18 @@ A quick start guide to get KubeVirt up and running inside Vagrant.
...
+Fedora 25 as it ships with Go 1.7.5.
+
+A helpful tool to dynamically manage multiple versions of Go is
+[gimme](https://github.com/travis-ci/gimme)
+
+```bash
+    eval "$(curl -sL \
+    https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | \
+    GIMME_GO_VERSION=stable bash)"
+```
```
Comment (berrange) [inline docs/getting-started.md]: Suggesting people to run arbitrary shell script code hosted on a remote website is *terrible* security practice. Please don't do this.

File: docs/getting-started.md [hunk 2]
```diff
@@ -26,6 +26,18 @@ A quick start guide to get KubeVirt up and running inside Vagrant.
     sudo dnf install golang
 ```
 
+**Note:** Some code within k8s.io/client-go and k8s.io/apimachinery uses
+features from the Go standard libaries introduced in version 1.8. This affects
+Fedora 25 as it ships with Go 1.7.5.
```
Comment (berrange) [inline docs/getting-started.md]: I don't think we need to call out Fedora 25 particularly here - it affects basically every distro that isn't a very recent release, or able to pull in GO libraries via some add on repository. It is sufficient to document that you need to be running a distro with Go 1.8.  I'd suggest we just have a section right at the start of the file where we list any pre-requisites we have for development

File: docs/getting-started.md (hunk 1 above)
Comment (stu-gott) [inline docs/getting-started.md]: This is the exact same command listed on the gimme README.md landing page, which users will find and execute anyways. The link in question of course points to openly available content hosted directly on github. Either you trust the tool or you don't.

File: docs/getting-started.md [hunk 3]
```diff
@@ -26,6 +26,11 @@ A quick start guide to get KubeVirt up and running inside Vagrant.
     sudo dnf install golang
 ```
 
+**Note:** Some code within k8s.io/client-go and k8s.io/apimachinery uses
+features from the Go standard libaries introduced in version 1.8.
+
+If needed, a helpful tool to dynamically manage multiple versions of Go is
+[gimme](https://github.com/travis-ci/gimme)
```
Comment (admiyo) [inline docs/getting-started.md]: Awesome.  Thanks

**Pull request merged**